comp/otelcol/otlp/collector.go

```diff
@@ -114,10 +114,11 @@ func getComponents(s serializer.MetricSerializer, logsAgentChannel chan *message
 		errs = append(errs, err)
 	}
 
-	processors, err := processor.MakeFactoryMap(
-		batchprocessor.NewFactory(),
-		infraattributesprocessor.NewFactory(),
```

Comment:
Looks like for serverless, we'll now no longer have this `infraattributesprocessor`. What are we losing when that processor goes away?

Reply:
The `infraattributesprocessor` was only [recently added](https://github.com/DataDog/datadog-agent/pull/25039) as a no-op processor, this is the first implementation of it so no functionality will be lost for serverless. This is only intended for use with the new otel-agent binary for now

Reply:
Okay thanks, that makes sense! Just so I understand, what will this processor do once it's completely written? Is it something that we'll want to support in AWS Lambda? The last question is as much for you as it is for me.

If we don't want it in serverless, then I want to make sure that we don't add extra code to our binary just for it to all be ignored during runtime. It's really important for us to keep the size of our binary down. So, as you work on this, you might want to consider using build tags to compile it out of the serverless binary all together. 

Let me know if you'll need a hand with any of this, happy to help.

Reply:
For example, this PR increases the size of our binary by about 16KB but adds no benefit. That's a miniscule amount, so I'm not real worried about it. But, it's something we'll want to keep an eye on.